### PR TITLE
Add spellcheck delimiter for RST directives

### PR DESCRIPTION
**Pull request by samjwu**
Goal: to allow RST directives

Examples:

```
.. doxygenenum::

.. doxygenstruct::
```

### CODE REVIEW

**Comment by samjwu** [comment]
re: https://github.com/ROCmSoftwarePlatform/hipTensor/actions/runs/6909405296/job/18800679016?pr=150

**Comment by saadrahim** [comment]
Can you separate the delimiters for rst from the word list update?

**Comment by samjwu** [comment]
> Can you separate the delimiters for rst from the word list update?

https://github.com/RadeonOpenCompute/rocm-docs-core/pull/527

File: .spellcheck.yaml [hunk 1]
```diff
@@ -12,7 +12,10 @@ matrix:
   - pyspelling.filters.context:
       context_visible_first: true
       delimiters:
-      # ReST: code blocks
+      # RST directives
+      - open: '^\.{2}[ ]+[a-z-]+[:]{2}'
```
Comment (saadrahim) [inline .spellcheck.yaml:16]: @neon60 how do you think this will interact with the other code block sections that were added?

Comment (neon60) [reply .spellcheck.yaml:16]: I am testing locally. The code block check also not working correctly, I need more time to be able to answer the question.

**Comment by neon60** [comment]
Find a better way to handle the ReST of spellcheck, we should probably just close this PR. 
We will discuss this during the today's standup.

**Comment by samjwu** [comment]
https://github.com/RadeonOpenCompute/rocm-docs-core/pull/530

**Pull request closed without merging**